### PR TITLE
Support items on promo & prequal request

### PR DESCRIPTION
**Pull request by wangdaliu**

### CODE REVIEW

File: affirm/src/main/java/com/affirm/android/Affirm.java [hunk 1]
```diff
@@ -568,6 +587,21 @@ public static void configureWithAmount(@NonNull final AffirmPromotionButton prom
         configureWithAmount(promotionButton, null, null, amount, showCta);
     }
 
+    /**
+     * Write the as low as span (text and logo) on a AffirmPromoLabel
+     *
+     * @param promotionButton AffirmPromotionButton to show the promo message
+     * @param amount          (Float) eg 112.02 as $112 and ¢2
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/Affirm.java:594]: Really a BigDecimal

File: affirm/src/main/java/com/affirm/android/Affirm.java [hunk 2]
```diff
@@ -583,6 +617,23 @@ public static void configureWithAmount(@NonNull final AffirmPromotionButton prom
         configureWithAmount(promotionButton, promoId, null, amount, showCta);
     }
 
+    /**
+     * Write the as low as span (text and logo) on a AffirmPromoLabel
+     *
+     * @param promotionButton AffirmPromotionButton to show the promo message
+     * @param promoId         the client's modal id
+     * @param amount          (Float) eg 112.02 as $112 and ¢2
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/Affirm.java:625]: Same

File: affirm/src/main/java/com/affirm/android/Affirm.java [hunk 3]
```diff
@@ -599,6 +650,24 @@ public static void configureWithAmount(@NonNull final AffirmPromotionButton prom
         configureWithAmount(promotionButton, null, pageType, amount, showCta);
     }
 
+    /**
+     * Write the as low as span (text and logo) on a AffirmPromoLabel
+     *
+     * @param promotionButton AffirmPromotionButton to show the promo message
+     * @param pageType        need to use one of "banner, cart, category, homepage, landing,
+     *                        payment, product, search"
+     * @param amount          (Float) eg 112.02 as $112 and ¢2
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/Affirm.java:659]: same

File: affirm/src/main/java/com/affirm/android/Affirm.java [hunk 4]
```diff
@@ -614,6 +683,26 @@ public static void configureWithAmount(@NonNull final AffirmPromotionButton prom
...
+    }
+
+    /**
+     * Write the as low as span (text and logo) on a AffirmPromoLabel
+     *
+     * @param promotionButton AffirmPromotionButton to show the promo message
+     * @param promoId         the client's modal id
+     * @param pageType        need to use one of "banner, cart, category, homepage, landing,
+     *                        payment, product, search"
+     * @param amount          (Float) eg 112.02 as $112 and ¢2
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/Affirm.java:696]: same

File: affirm/src/test/java/com/affirm/android/PromoRequestTest.java [hunk 5]
```diff
@@ -46,7 +46,7 @@ public void onFailure(@NonNull AffirmException exception) {
         };
 
         final PromoRequest affirmPromoRequest =
-                new PromoRequest(null, null, BigDecimal.valueOf(1100.0), false, AffirmColor.AFFIRM_COLOR_TYPE_BLUE, AffirmLogoType.AFFIRM_DISPLAY_TYPE_LOGO, true, callback);
+                new PromoRequest(null, null, BigDecimal.valueOf(1100.0), false, AffirmColor.AFFIRM_COLOR_TYPE_BLUE, AffirmLogoType.AFFIRM_DISPLAY_TYPE_LOGO, true, null, callback);
```
Comment (kevindavmoore) [inline affirm/src/test/java/com/affirm/android/PromoRequestTest.java:49]: Do you want a test with items in it?

Comment (wangdaliu) [reply affirm/src/test/java/com/affirm/android/PromoRequestTest.java]: Added

File: affirm/src/main/java/com/affirm/android/Affirm.java (hunk 4 above)
Comment (wangdaliu) [reply affirm/src/main/java/com/affirm/android/Affirm.java]: Fixed

File: affirm/src/main/java/com/affirm/android/Affirm.java (hunk 3 above)
Comment (wangdaliu) [reply affirm/src/main/java/com/affirm/android/Affirm.java]: Fixed

File: affirm/src/main/java/com/affirm/android/Affirm.java (hunk 2 above)
Comment (wangdaliu) [reply affirm/src/main/java/com/affirm/android/Affirm.java]: Fixed

File: affirm/src/main/java/com/affirm/android/Affirm.java (hunk 1 above)
Comment (wangdaliu) [reply affirm/src/main/java/com/affirm/android/Affirm.java]: Fixed

**Comment by vcello** [comment]
Reviewed and approved

**Pull request merged**